cardano-cli/src/Cardano/CLI/EraBased/Run/CreateTestnetData.hs

```diff
@@ -306,18 +308,61 @@ runGenesisCreateTestNetDataCmd Cmd.GenesisCreateTestNetDataCmdArgs
       delegAddrs stuffedUtxoAddrs shelleyGenesis
 
   -- Write genesis.json file to output
-  liftIO $ LBS.writeFile (outputDir </> "genesis.json") $ Aeson.encode shelleyGenesis'
+  liftIO $ LBS.writeFile (outputDir </> "conway-genesis.json") $ Aeson.encode conwayGenesis'
+  liftIO $ LBS.writeFile (outputDir </> "shelley-genesis.json") $ Aeson.encode shelleyGenesis'
+  liftIO $ LBS.writeFile (outputDir </> "alonzo-genesis.json") $ Aeson.encode alonzoGenesis
   where
     genesisDir = outputDir </> "genesis-keys"
     delegateDir = outputDir </> "delegate-keys"
     drepsDir = outputDir </> "drep-keys"
     utxoKeysDir = outputDir </> "utxo-keys"
     poolsDir = outputDir </> "pools-keys"
     stakeDelegatorsDir = outputDir </> "stake-delegators"
-    numStakeDelegators = case stakeDelegators of OnDisk n -> n; Transient n -> n
     mkDelegationMapEntry :: Delegation -> (L.KeyHash L.Staking L.StandardCrypto, L.PoolParams L.StandardCrypto)
     mkDelegationMapEntry d = (dDelegStaking d, dPoolParams d)
 
+    addDRepsToConwayGenesis :: [VerificationKey DRepKey] -> [VerificationKey StakeKey]
+                            -> L.ConwayGenesis L.StandardCrypto -> L.ConwayGenesis L.StandardCrypto
+    addDRepsToConwayGenesis dRepKeys stakingKeys conwayGenesis =
+      conwayGenesis { L.cgDelegs = delegs (zip stakingKeys (case dRepKeys of [] -> []; _ -> cycle dRepKeys))
+                    , L.cgInitialDReps = initialDReps dRepKeys
+                    }
+
+      where
+      delegs :: [(VerificationKey StakeKey, VerificationKey DRepKey)] -> ListMap (L.Credential L.Staking L.StandardCrypto) (L.Delegatee L.StandardCrypto)
+      delegs = ListMap.fromList . map (bimap verificationKeytoStakeCredential (L.DelegVote . L.DRepCredential . verificationKeyToDRepCredential))
+
+      initialDReps :: [VerificationKey DRepKey] -> ListMap (L.Credential L.DRepRole L.StandardCrypto) (L.DRepState L.StandardCrypto)
+      initialDReps = ListMap.fromList . map (\c -> ( verificationKeyToDRepCredential c
+                                                   , L.DRepState { L.drepExpiry = EpochNo 1000
+                                                                 , L.drepAnchor = SNothing
+                                                                 , L.drepDeposit = L.Coin 100000000
```

Comment:
Question regarding the expiry and the deposit: aren't these values possibly defined in the genesis template? In this case we would probably want to reuse the template's value here.

Reply:
`initialDReps` is defined as `mempty` in the template.
I got inspiration from this: https://github.com/IntersectMBO/cardano-ledger/blob/4ad68470b01be95c7dd523c839e3ef4e84d026a8/eras/conway/impl/test/data/conway-genesis.json
But it doesn't have one specific amount, it has different for each entry. I imagine for testing they probably want them to not expire, in principle, so I gave it a large value for expire, and `₳100` for the deposit.

Reply:
OK, sounds good to me :+1: